### ISSUE TITLE
Add "Use CurrentPageInfo when possible" module

### ISSUE THREAD

**Issue by ChristopherJennings** [Kentico/KInspector#178]
### Motivation

When accessing data for the current page, `CurrentPageInfo` had better performance than `CurrentDocument` because it is already loaded and available within the page request. When the needed data is available on `CurrentPageInfo` that should be used instead of `CurrentDocument`. 

### Proposed solution

List all locations where `CurrentPageInfo` can be used instead of `CurrentDocument`. A good initial version will do the following:
- Find uses of `CurrentDocument` in the following areas:
  - Transformations
  - Web part properties
  - Page Template Layouts
  - HEAD section of master template
- Determine if the data being accessed on `CurrentDocument` is available on `CurrentPageInfo` via a 1-to-1 property check.
- Report instances where `CurrentPageInfo` can be used (show as error)
- Report instances where `CurrentDocument` is used but can't be replaced (show as warning)

### Additional context

Original discussion: #6 
Should be merged into "Transformation Performance Analysis"
Blocked by #232

**Comment by kentico-ericd** [comment]
As we are now mainly supporting K13, this is not applicable
